### PR TITLE
Creating a Song with Tone.js

### PR DESCRIPTION
**Pull request by nerap08**
Thanks for making a PR! Please fill in the below:

Name: Paul Trusov

Do you run a club? Yes, I run the Hack Club at the Montgomery Blair High School

What relevant experiences do you have to help you write this Jam? I have been using javascript in different projects for a few years now, I also worked on a project using Tone.js

Will you be available on Slack to provide support to Hack Clubbers who run into issues while going through your Jam? Certainly, I would be happy to help out other Hack Clubbers.

### CODE REVIEW

**Comment by Arpan-206** [comment]
The file is JS, can you change that to md?

**Comment by nerap08** [comment]
My bad I didn't notice it was in JS, I fixed it now.

File: jams/Creating a Song with Tone.js.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+What makes this workshop fun or interesting for Club Members?
+Club members will have the chance to their own songs that they would get to share with one another. 
+  
+What platforms will be supported (i.e. MacOS, Windows, Chromebook, Mobile, Browser, etc)?
+Any device that has access to Replit.
+
+How will you allow Club Leaders to add their own project to the Jam presentation (giving them a sense of ownership over the meeting)?
+Club leaders could choose the inspiration for the songs created during the Jam.
+  
+General outline:
```
Comment (Arpan-206) [inline jams/Creating a Song with Tone.js.md]: Can you add specific timelines for each?

File: jams/Creating a Song with Tone.js.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+Club leaders could choose the inspiration for the songs created during the Jam.
+  
+General outline: 
+1. Introduction to Tone.js and its capabilities in music production.
+2. Setting up a Replit project with Tone.js library included.
+3. Understanding the basics of music synthesis, including instruments, oscillators, and filters.
+4. Adding beats and rhythms using Tone.js's drum machine or sampler.
+5. Manipulating and modulating sounds with different effects.
+6. Fine-tuning the song by adjusting tempo, volume, and other parameters.
+7. Sharing the created songs within the Jam session.
```
Comment (Arpan-206) [inline jams/Creating a Song with Tone.js.md]: A lot of the questions later haven't been answered.

**Review by Arpan-206 — CHANGES_REQUESTED** [review]
Missing answers

**Comment by nerap08** [comment]
I added a timeline, but I can't seem to find the questions which I haven't answered yet.

**Comment by Arpan-206** [comment]
> I added a timeline, but I can't seem to find the questions which I haven't answered yet.

I suggest having a look at some merged PRs.

**Comment by nerap08** [comment]
I changed my file into the structure of the "Add Kaboom" jam file.

File: jams/Creating a Song with Tone.js.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+3) The Jam will show members how music can be combined with coding. 
+
+#### What Club Members will walk away with (both in terms of knowledge and in terms of product)
+1) Members will have a better understanding of Javascript.
+2) Members will know how to use tone.js
+3) Members will have their own song made in replit.
+#### What makes this workshop fun or interesting for Club Members?
+Not many people have heard of making songs with coding, having the opportunity to do so can be an interesting experience to them.
+#### What platforms will be supported (i.e. MacOS, Windows, Chromebook, Mobile, Browser, etc)?
+Any device with access to replicate.
```
Comment (miguelaenlle) [inline jams/Creating a Song with Tone.js.md:43]: Typo: Replicate should be "Replit"

Comment (miguelaenlle) [inline jams/Creating a Song with Tone.js.md]: Typo: Change "replicate" to "replit"

**Review by miguelaenlle — CHANGES_REQUESTED** [review]

File: jams/Creating a Song with Tone.js.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+_Customization opportunity:_ Participants will have the opportunity to express themselves by composing their own unique songs. They can experiment with different instruments, rhythms, and melodies to create a song.
+
+_Platform Limitations:_ No limitations as long as members have access to a replit.
+
+_Time estimation:_ 1 hour
+
+### Breakdown
+1. 5 minutes: Introduction to Tone.js and its capabilities in music production.
+2. 10 minutes: Setting up a Replit project with Tone.js library included.
+3. 5 minutes: Understanding the basics of music synthesis, including instruments, oscillators, and filters.
```
Comment (miguelaenlle) [inline jams/Creating a Song with Tone.js.md]: Go into more detail on the basics of music synthesis, and how they all connect to enable you to produce a song.

**Review by miguelaenlle — CHANGES_REQUESTED** [review]

**Comment by nerap08** [comment]
I have made those two changes to the file, do they look good?

**Comment by miguelaenlle** [comment]
Yup, looks good @nerap08

**Review by miguelaenlle — APPROVED** [review]

**Review by linkai101 — CHANGES_REQUESTED** [review]
I love projects that involve music and code, but this seems a bit too simple for a jam imo, more like a tutorial for Tone.js rather than a jam.

So far, it's a lot like [muse](https://workshops.hackclub.com/muse/), which—don't get me wrong—is a great workshop, but given the standards for a jam, I'd recommend adding on to this idea with a specific project in mind, like a [synth pad](https://workshops.hackclub.com/synth/) or a [piano](https://workshops.hackclub.com/tunes/). (both already existing workshops) Also, there doesn't seem to be a visual component, so having that could help.  Hope that makes sense!

**Comment by nerap08** [comment]
Hey Linkai, to make the Jam more complex we can let the members make a melody and then have them code it so that the speed of the rhythm and level of the pitch can be changed by the users when running it. This would still allow the members to have the freedom to make their own song but it would also allow them to hear different variations of it each time they run it. Additionally, to add a visual component we can change the color of the background each time a new note is played. This would make it look prettier and it would allow members to see the pace of their song.

**Review by Arpan-206 — APPROVED** [review]
LGTM!

**Comment by SerenityUX** [comment]
Great feedback Linkai!

> I love projects that involve music and code, but this seems a bit too simple for a jam imo, more like a tutorial for Tone.js rather than a jam.
> 
> So far, it's a lot like [muse](https://workshops.hackclub.com/muse/), which—don't get me wrong—is a great workshop, but given the standards for a jam, I'd recommend adding on to this idea with a specific project in mind, like a [synth pad](https://workshops.hackclub.com/synth/) or a [piano](https://workshops.hackclub.com/tunes/). (both already existing workshops) Also, there doesn't seem to be a visual component, so having that could help. Hope that makes sense!

I think with some work, this can be made into a SUPER FUN JAM that I would love to use with my club!

How might Hack Clubbers be able work together to make a really awesome symphony?

**Comment by nerap08** [comment]
With the ability to control tempo without much work, clubs can effortlessly combine different melodies, into harmony. This ability would enable groups to join their projects together, making an awesome symphony.

**Comment by linkai101** [comment]
Yeah creativity and uniqueness definitely isn't an issue here. You know, this kinda reminds me of Google's [blob opera](https://artsandculture.google.com/experiment/blob-opera/AAHWrq360NcGbw?hl=en). Making that with Tone.js would be awesome--and the color changing def helps.

**Review by linkai101 — APPROVED** [review]
This looks good to me! Though I agree with Thomas that it may need some work, that can be addressed in the next phase and I definitely look forward to this jam. :)

**Review by sahitid — CHANGES_REQUESTED** [review]
hi! love the idea, but this very closely mimics existing workshops on the hack club workshops page (such as synth and piano). we're really looking for something new with these bounty proposals and i'm afraid this is not meeting the mark.

**Pull request closed without merging**